packages/create-cloudflare/src/common.ts

```diff
@@ -128,7 +130,7 @@ export const runDeploy = async (ctx: PagesGeneratorContext) => {
 	const deployCmd = [
 		...baseDeployCmd,
 		// Important: the following assumes that all framework deploy commands terminate with `wrangler pages deploy`
-		...(ctx.framework?.commitMessage && !insideGitRepo
+		...(ctx.framework?.commitMessage && (!insideGitRepo || ctx.insideMonorepo)
```

Comment:
Based on clarification in chat, can you rename `insideMonorepo` to `insidePreExistingRepo` or something similar to match what it actually is